### PR TITLE
Union non-spatial scales in repeat.

### PR DESCRIPTION
**Pull request by domoritz**
screen shot 2017-04-18 at 09 34 54

### CODE REVIEW

File: src/compile/repeat.ts [hunk 1]
```diff
@@ -144,8 +177,21 @@ export class RepeatModel extends Model {
   }
 
   public parseLegend() {
+    const legendComponent = this.component.legends = {};
+
     for (const child of this.children) {
       child.parseLegend();
+
+      // TODO: correctly implement independent axes
```
Comment (kanitw) [inline src/compile/repeat.ts]: independent legend*

File: src/compile/repeat.ts [hunk 2]
```diff
@@ -144,8 +177,21 @@ export class RepeatModel extends Model {
...
+      if (true) { // if shared/union scale
+        keys(child.component.legends).forEach(function(channel) {
```
Comment (kanitw) [inline src/compile/repeat.ts]: // TODO: need to decide if the legend are not compatible. Do we still merge?

File: src/compile/repeat.ts [hunk 3]
```diff
@@ -120,10 +122,41 @@ export class RepeatModel extends Model {
...
 
-    this.component.scales = {};
+    const scaleComponent: Dict<VgScale> = this.component.scales = {};
 
     this.children.forEach(function(child) {
       child.parseScale();
+
+      // FIXME(#1602): correctly implement independent scale
+      // Also need to check whether the scales are actually compatible, e.g. use the same sort or throw error
+      if (true) { // if shared/union scale
```
Comment (kanitw) [inline src/compile/repeat.ts]: // TODO: need to decide if the scales are not compatible. Do we still merge?

File: src/compile/repeat.ts (hunk 1 above)
Comment (domoritz) [inline src/compile/repeat.ts]: Done

**Pull request merged**